codec-http/src/test/java/io/netty/handler/codec/http/websocketx/extensions/compression/PerFrameDeflateDecoderTest.java

```diff
@@ -122,4 +119,40 @@ public void testCompressedEmptyFrame() {
         assertEquals(0, uncompressedFrame.content().readableBytes());
         uncompressedFrame.release();
     }
+
+    @Test
+    public void testCompressionSkip() {
+        EmbeddedChannel encoderChannel = new EmbeddedChannel(
+                ZlibCodecFactory.newZlibEncoder(ZlibWrapper.NONE, 9, 15, 8));
+        EmbeddedChannel decoderChannel = new EmbeddedChannel(new PerFrameDeflateDecoder(false, ALWAYS_SKIP));
+        byte[] payload = new byte[300];
+        random.nextBytes(payload);
+        BinaryWebSocketFrame binaryFrame = new BinaryWebSocketFrame(true,
+                WebSocketExtension.RSV3, Unpooled.wrappedBuffer(payload));
+
+        // execute
+        encoderChannel.writeOutbound(binaryFrame.copy());
+        BinaryWebSocketFrame outboundFrame = encoderChannel.readOutbound();
+
+        // test
+        assertNotNull(outboundFrame);
+        assertNotNull(outboundFrame.content());
+        assertArrayEquals(payload, ByteBufUtil.getBytes(outboundFrame.content()));
+        assertEquals(WebSocketExtension.RSV3, outboundFrame.rsv());
+        assertTrue(outboundFrame.release());
+
+        decoderChannel.writeInbound(binaryFrame);
+
+        BinaryWebSocketFrame inboundBinaryFrame = decoderChannel.readInbound();
+
+        assertNotNull(inboundBinaryFrame);
+        assertNotNull(inboundBinaryFrame.content());
+        assertArrayEquals(payload, ByteBufUtil.getBytes(inboundBinaryFrame.content()));
+        assertEquals(WebSocketExtension.RSV3, inboundBinaryFrame.rsv());
+
+        assertTrue(inboundBinaryFrame.release());
+
+        assertEquals(0, binaryFrame.refCnt());
```

Comment:
call `channel.finish()` and assert return value

Reply:
Done.